great_expectations/core/id_dict.py

```diff
@@ -1,6 +1,10 @@
 import hashlib
 import json
-from typing import Any, Optional, Set
+from typing import Any, Set, TypeVar, Union
+
+from great_expectations.core.util import convert_to_json_serializable
+
+T = TypeVar("T")
```

Comment:
What is the purpose of this?

Reply:
I see docs on it [here](https://docs.python.org/3/library/typing.html#generics), but I don't understand how it applies to this specific situation.

Reply:
@NathanFarmer This is something very useful, which @billdirks introduced in parameter container module.  It enables us to replace the use of (unhelpful, only last resort) type declaration `Any` with an actual type that says that whatever type that will come in to an input argument will be returned.  This is useful because as part of conversion of a nested iterable structure (dict, list, etc.), any kind of a sub-structure can be encountered; this `TypeVar` representation helps us to say formally that this is happening.  Thanks!